Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml

```diff
@@ -949,3 +949,48 @@
   - type: Tag
     tags:
       - HideContextMenu
+
+- type: entity
+  name : ion rifle bolt
+  id: BulletIonRifle
+  parent: BaseBulletTrigger
+  noSpawn: true
+  components:
+  - type: Reflective
+    reflective:
+    - Energy
+  - type: FlyBySound
+    sound:
+      collection: EnergyMiss
+      params:
+        volume: 5
+  - type: Sprite
+    sprite: Effects/emp.rsi
+    layers:
+    - state: emp_pulse
+      shader: unshaded
+  - type: Physics
+  - type: Fixtures
+    fixtures:
+      projectile:
+        shape:
+          !type:PhysShapeAabb
+          bounds: "-0.15,-0.3,0.15,0.3"
+        hard: false
+        mask:
+        - Impassable
+        - BulletImpassable
+      fly-by: *flybyfixture
+  - type: Ammo
+  - type: Projectile
+    impactEffect: BulletImpactEffectDisabler
+    damage:
+      types:
+        Shock: 5
+    soundHit:
+      collection: WeakHit
+    forceSound: true
+  - type: EmpOnTrigger
+    range: 2
+    energyConsumption: 200
+    disableDuration: 5
```

Comment:
This shouldn't be relying on triggersystem, it should just be its own component for EMPs.

Reply:
What do you mean?

Reply:
This has nothing to do with voice triggers or grenades which is what trigger is for.

Reply:
then it has copy paste EmpOnSpawn bloat code

Reply:
Sloth RPG's, EMP grenades shot from ship guns, every explosive projectile, all do this

Reply:
Grenades are different, I'll just fix it.